include/glow/Importer/CommonOperatorLoader.h

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GLOW_IMPORTER_COMMONOPERATORLOADER_H
+#define GLOW_IMPORTER_COMMONOPERATORLOADER_H
+
+#include "glow/Importer/ProtobufLoader.h"
+
+#include "glow/Base/Tensor.h"
+#include "glow/Graph/Graph.h"
+
+#include "llvm/ADT/ArrayRef.h"
+
+#include <functional>
+#include <string>
+#include <unordered_map>
+
+namespace glow {
+
+/// Contains loaders for operators, which are common to ONNX and Caffe2 formats.
+/// Every loader method adds necessary nodes to property G_, which is inherited
+/// from ProtobufLoader class, therefore modifying the class instance itself.
+template <typename OpType, typename AttrType>
+class CommonOperatorLoader : public ProtobufLoader {
```

Comment:
I do not see a lot of benefits of inheritance in this case.

One of the approaches would be to have a templated function with the switch statement which returns true/false (true in case common case was handled).
Then onnx/caffe2 impls calls into a common function with the specific types and handles false case separately.

Reply:
Inheritance is required, because creating Relu changes Graph and uses things like `getOrCreateNodeByName()`.

Reply:
Bummer...